### PR TITLE
network/wifi_manager: Add register, unregister callback API

### PR DESCRIPTION
**Pull request by SeonghoByeon**
- By adding these APIs, applications on TizenRT can get wi-fi events max 3 by default. 
- Note that the 0th index of the callback array is allocated by initializing Wi-Fi Manager, wifi_manager_init.

### CODE REVIEW

File: framework/src/wifi_manager/wifi_manager.c [hunk 1]
```diff
@@ -1285,60 +1301,66 @@ wifi_manager_result_e _handler_on_scanning_state(_wifimgr_msg_s *msg)
...
+		wifi_manager_cb_s *cbk = g_manager_info.cb[i];
+		if (!cbk) {
+			continue;
+		}
+		switch (evt) {
+		case CB_STA_CONNECTED:
+			nvdbg("[WM] call sta connect success event\n");
 #ifdef CONFIG_ENABLE_IOTIVITY
-		__tizenrt_manual_linkset("del");
+			__tizenrt_manual_linkset("gen");
```
Comment (thapav) [inline framework/src/wifi_manager/wifi_manager.c]: Please correct indentation.

Comment (thapav) [inline framework/src/wifi_manager/wifi_manager.c]: Please correct indentation.

File: framework/src/wifi_manager/wifi_manager.c [hunk 2]
```diff
@@ -1285,60 +1301,66 @@ wifi_manager_result_e _handler_on_scanning_state(_wifimgr_msg_s *msg)
...
+	#ifdef CONFIG_ENABLE_IOTIVITY
+			__tizenrt_manual_linkset("del");
+	#endif
+			cbk->sta_disconnected(WIFI_MANAGER_DISCONNECT);
+			break;
+		case CB_STA_RECONNECTED:
+			nvdbg("[WM] call sta disconnect event\n");
 #ifdef CONFIG_ENABLE_IOTIVITY
-		__tizenrt_manual_linkset("del");
+			__tizenrt_manual_linkset("del");
```
Comment (thapav) [inline framework/src/wifi_manager/wifi_manager.c]: same as above

File: framework/src/wifi_manager/wifi_manager.c [hunk 3]
```diff
@@ -1586,3 +1608,43 @@ wifi_manager_result_e wifi_manager_get_stats(wifi_manager_stats_s *stats)
...
+{
+	wifi_manager_result_e res = WIFI_MANAGER_FAIL;
+
+	LOCK_WIFIMGR;
+	// g_manager_info.cb[0] is assigned to the callback which is registered by wifi_manager_init
+	if (g_manager_info.cb[0] == NULL) {
+		UNLOCK_WIFIMGR;
+		return WIFI_MANAGER_DEINITIALIZED;
+	}
+	int i = 1;
```
Comment (thapav) [inline framework/src/wifi_manager/wifi_manager.c]: you could add integer declaration at the start of wifi_manager_register_cb API along with other declarations & initialize it in the for loop

File: framework/src/wifi_manager/wifi_manager.c (hunk 1 above)
Comment (thapav) [inline framework/src/wifi_manager/wifi_manager.c]: please correct indentation

Comment (thapav) [inline framework/src/wifi_manager/wifi_manager.c]: please correct indentation

File: framework/src/wifi_manager/wifi_manager.c (hunk 2 above)
Comment (thapav) [inline framework/src/wifi_manager/wifi_manager.c]: same as above

File: framework/src/wifi_manager/wifi_manager.c (hunk 3 above)
Comment (thapav) [inline framework/src/wifi_manager/wifi_manager.c]: declaration could be added at the start of wifi_manager_register_cb API with rest of the declarations
Initialization in the for loop

File: framework/src/wifi_manager/wifi_manager.c (hunk 2 above)
Comment (thapav) [inline framework/src/wifi_manager/wifi_manager.c]: same as above

File: framework/src/wifi_manager/wifi_manager.c (hunk 3 above)
Comment (thapav) [inline framework/src/wifi_manager/wifi_manager.c]: declaration could be added at the start of wifi_manager_register_cb API with rest of the declarations
Initialization in the for loop

File: framework/src/wifi_manager/wifi_manager.c (hunk 1 above)
Comment (SeonghoByeon) [reply framework/src/wifi_manager/wifi_manager.c]: Current indentation is correct, considering below lines. Thanks.

File: framework/src/wifi_manager/wifi_manager.c (hunk 2 above)
Comment (SeonghoByeon) [reply framework/src/wifi_manager/wifi_manager.c]: Current indentation is correct, considering below lines. Thanks.

File: framework/src/wifi_manager/wifi_manager.c (hunk 3 above)
Comment (SeonghoByeon) [reply framework/src/wifi_manager/wifi_manager.c]: Modification completed. Thanks.

**Comment by seinfra** [comment]
Target : [4205fb08d9e8334c82ee34a5bb0f44be79f19d0b] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [4205fb08d9e8334c82ee34a5bb0f44be79f19d0b] - Code Rule Check OK.

File: framework/include/wifi_manager/wifi_manager.h [hunk 4]
```diff
@@ -283,7 +305,7 @@ wifi_manager_result_e wifi_manager_get_info(wifi_manager_info_s *info);
  * @brief Connect to an access point.
  * @details @b #include <wifi_manager/wifi_manager.h>
  * @param[in] config ssid, passphrase, authentication type, and cryto type of the access point which the wi-fi interface connect to.
- * @param[in] reconn_config reconnect type, interval, minimum or maximun intervalis set
+ * @param[in] reconn_config reconnect type, interval, minimum or maximun interval is set
```
Comment (shadaksharikp) [inline framework/include/wifi_manager/wifi_manager.h]: correct spell mistakes
maximun => maximum

File: framework/include/wifi_manager/wifi_manager.h [hunk 5]
```diff
@@ -293,7 +315,7 @@ wifi_manager_result_e wifi_manager_connect_ap_config(wifi_manager_ap_config_s *c
  * @brief Connect to an access point.
  * @details @b #include <wifi_manager/wifi_manager.h>
  * @param[in] config ssid, passphrase, authentication type, and cryto type of the access point which the wi-fi interface connect to.
- * @param[in] conn_config reconnect type, interval, minimum or maximun intervalis set 
+ * @param[in] reconn_config reconnect type, interval, minimum or maximun intervals is set
```
Comment (shadaksharikp) [inline framework/include/wifi_manager/wifi_manager.h]: correct spell mistakes
maximun => maximum

**Comment by seinfra** [comment]
Target : [81acde787a03e0f319b60e7abf8ace50acca7a2f] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [81acde787a03e0f319b60e7abf8ace50acca7a2f] - Code Rule Check OK.

File: framework/include/wifi_manager/wifi_manager.h [hunk 6]
```diff
@@ -283,7 +305,7 @@ wifi_manager_result_e wifi_manager_get_info(wifi_manager_info_s *info);
  * @brief Connect to an access point.
  * @details @b #include <wifi_manager/wifi_manager.h>
  * @param[in] config ssid, passphrase, authentication type, and cryto type of the access point which the wi-fi interface connect to.
```
Comment (shadaksharikp) [inline framework/include/wifi_manager/wifi_manager.h]: Spell mistakes needs to be corrected
cryto => crypto

File: framework/include/wifi_manager/wifi_manager.h [hunk 7]
```diff
@@ -293,7 +315,7 @@ wifi_manager_result_e wifi_manager_connect_ap_config(wifi_manager_ap_config_s *c
...
```
Comment (shadaksharikp) [inline framework/include/wifi_manager/wifi_manager.h]: Spell mistakes needs to be corrected
cryto => crypto

**Comment by seinfra** [comment]
Target : [75e6ba6647b824cf9d1eae8d7065c917e881e8e1] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [75e6ba6647b824cf9d1eae8d7065c917e881e8e1] - Code Rule Check (C++) OK.

File: framework/include/wifi_manager/wifi_manager.h (hunk 6 above)
Comment (SeonghoByeon) [reply framework/include/wifi_manager/wifi_manager.h]: Modification completed. Thanks.

File: framework/include/wifi_manager/wifi_manager.h (hunk 7 above)
Comment (SeonghoByeon) [reply framework/include/wifi_manager/wifi_manager.h]: Modification completed. Thanks.

**Pull request merged**